### PR TITLE
Accept current password in changeStrongholdPassword

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Accept current password in changeStrongholdPassword to have it more clear

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account_manager/operations/stronghold.rs [hunk 1]
```diff
@@ -19,11 +19,13 @@ impl AccountManager {
...
     pub async fn change_stronghold_password(
         &self,
-        password: &str,
+        current_password: &str,
+        new_password: &str,
         keys_to_re_encrypt: Option<Vec<Vec<u8>>>,
     ) -> crate::Result<()> {
         if let SecretManager::Stronghold(stronghold) = &mut *self.secret_manager.write().await {
-            stronghold.change_password(password, keys_to_re_encrypt).await?;
+            stronghold.set_password(current_password).await?;
```
Comment (thibault-martinez) [inline src/account_manager/operations/stronghold.rs:27]: Why is this needed ?

Comment (Thoralf-M) [reply src/account_manager/operations/stronghold.rs:27]: Because the user gets an error if it wasn't set before and seems to be good to make it "atomic"

Comment (thibault-martinez) [reply src/account_manager/operations/stronghold.rs:27]: Fair enough, I'd argue it's not terrible if it actually errors if the password wasn't set but I don't mind that much, approving.

**Review by thibault-martinez — APPROVED** [review]

**Review by Tuditi — APPROVED** [review]
Looks good to me!
